python/triton/experimental/gluon/language/_semantic.py

```diff
@@ -57,11 +62,19 @@ def expand_dims(self, input: TensorTy, axis: int) -> TensorTy:
         handle = self.builder.create_expand_dims(input.handle, axis, ret_ty.to_ir(self.builder))
         return self.tensor(handle, ret_ty)
 
+    def join(self, a: TensorTy, b: TensorTy) -> TensorTy:
```

Comment:
JoinOp actually has a variety of possible result layouts. The type inference hook picks a default one but it would be nice to expose an optional result layout argument